### PR TITLE
web ide: security headers

### PR DESCRIPTION
**Pull request by bolek-wisniewski**
includes style fixes to hide terminal, debugger, and navigation widget.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by garyverhaegen-da** [comment]
Overall looks good to me. I would suggest adding the URL of the wiki page maintained by security above the security headers. I was also thinking, if you want to hide the terminal but can't, can you make it crash instead? What happens if you remove e.g. `bash` from the image?

File: web-ide/ide-server/Dockerfile [hunk 1]
```diff
@@ -11,10 +11,11 @@ RUN curl -sL https://deb.nodesource.com/setup_8.x | bash - &&\
         procps \
         nodejs \
         libxkbfile-dev \
-        libsecret-1-dev
+        libsecret-1-dev &&\
+    apt-get remove -y --allow-remove-essential apt
 
 USER sdk
-ENV VERSION=0.11.19 \
+ENV VERSION=0.12.16 \
```
Comment (neil-da) [inline web-ide/ide-server/Dockerfile]: Any chance we can move these two versions together, ideally call them DAML_VERSION (to be distinct from CODESERVER_VERSION). Actually combining them to a single version would be preferable.

File: web-ide/proxy/static/css/webide.main.css [hunk 2]
```diff
@@ -40,24 +40,38 @@ div.window-appicon {
...
-.monaco-menu-container .monaco-action-bar ul.actions-container li.action-item:nth-child(5), /*Keyboard shortcuts*/
-.monaco-menu-container .monaco-action-bar ul.actions-container li.action-item:nth-child(6), /*empty*/
-.monaco-menu-container .monaco-action-bar ul.actions-container li.action-item:nth-child(7), /*User Snippets*/
-.monaco-menu-container .monaco-action-bar ul.actions-container li.action-item:nth-child(8) { /*empty*/
+.global-activity { /*hide the entire global actions menu, this is mostly unwanted functionality*/
   display: none !important;
 }
 
+/*hide the commands widget, this is useful for file navigation but entering '>' allows you to execute any command*/
+.monaco-quick-open-widget {
```
Comment (neil-da) [inline web-ide/proxy/static/css/webide.main.css]: I'd rather leave this present and accept that it's a possibility to sneak stuff though. The file navigation is just too useful. Many of the commands (like converting case etc) are so standard dev features that without the quick open widget things aren't much fun.

File: web-ide/ide-server/Dockerfile (hunk 1 above)
Comment (bolek-wisniewski) [reply web-ide/ide-server/Dockerfile]: I think it's some annoyance with changing users...I could not use the arg value from the top of the file!

I'll try again, maybe I was doing something stupid.

File: web-ide/proxy/static/css/webide.main.css (hunk 2 above)
Comment (bolek-wisniewski) [reply web-ide/proxy/static/css/webide.main.css]: 👍

File: web-ide/ide-server/Dockerfile (hunk 1 above)
Comment (bolek-wisniewski) [reply web-ide/ide-server/Dockerfile]: found a work-around, apparently you have to redefine (but you don't have to set the default value again) the ARG. This is some oddity with defining ARG before FROM, it all gets wiped after the FROM

**Comment by bolek-wisniewski** [comment]
> Overall looks good to me. I would suggest adding the URL of the wiki page maintained by security above the security headers. I was also thinking, if you want to hide the terminal but can't, can you make it crash instead? What happens if you remove e.g. `bash` from the image?

I'm an ass @garyverhaegen-da  - Shayne's big image distracted my from your comment! (so I merged)

I'm not sure I know what wiki page you mean actually. In any case - this link probably shouldn't be in the public repo.
I'll think about the bash thing ... I'd like to be able to shell into this docker image to investigate.

**Comment by neil-da** [comment]
There is a setting in VS Code to change what the integrated terminal is. If you set it to `echo "No terminal available"` that may mostly hide it.

**Comment by bolek-wisniewski** [comment]
Thanks @neil-da and @garyverhaegen-da - I got it failing! #1087

**Pull request merged**